sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -98,6 +99,14 @@ object DateTimeUtils {
     sdf
   }
 
+  private val threadLocalTimeZones = new ThreadLocal[mutable.Map[String, TimeZone]] {
```

Comment:
As we won't go to update this map. We only need synchronization when putting the values. The content of the map can be shared between threads when reading. I am wondering if we need a local map for each thread.